File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.java
class Solution {
    public long[] solution(long[] numbers) {
        long[] answer = new long[numbers.length];
        for(int i = 0; i < numbers.length; i++) {
            if(numbers[i] % 2 == 0) {
                answer[i] = numbers[i]+1;
            }
            else {
                long bit = 1;
                while((numbers[i] & bit) != 0) {
                    bit <<= 1;
                    
                }
                numbers[i] = numbers[i] + bit - (bit >> 1);
                answer[i] = numbers[i];
                
            }
            
        }
        return answer;
    }
}

File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.js
function solution(numbers) {
    var answer = [];
    for (const number of numbers) {
        // 짝수는 다음 수랑 비트가 1개 다름
        if (number % 2 === 0) answer.push(number+1);
        else {
            let bin = '0' + number.toString(2);
            let index = bin.lastIndexOf('01');
            bin = bin.substring(0,index) + '10' + bin.substring(index+2);
            answer.push(parseInt(bin,2));
        }
    }
    return answer;
}